File: MSI-project/src/app/models/UserAccount.ts
import { Role } from "./role";

export class UserAccount {


  constructor(
    private _id: number,
    private _login: string,
    private _password: string,
    private _email: string,
    private _description: string,
    private _instructor: boolean,
    private _pricePerHour: number,
    private _role: Role
  ) {}

  public get role(): Role {
    return this._role;
  }
  public set role(value: Role) {
    this._role = value;
  }
  public get pricePerHour(): number {
    return this._pricePerHour;
  }
  public set pricePerHour(value: number) {
    this._pricePerHour = value;
  }
  public get instructor(): boolean {
    return this._instructor;
  }
  public set instructor(value: boolean) {
    this._instructor = value;
  }
  public get description(): string {
    return this._description;
  }
  public set description(value: string) {
    this._description = value;
  }
  public get email(): string {
    return this._email;
  }
  public set email(value: string) {
    this._email = value;
  }
  public get password(): string {
    return this._password;
  }
  public set password(value: string) {
    this._password = value;
  }
  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }

  public get login(): string {
    return this.login;
  }
  public set login(value: string) {
    this.login = value;
  }
}

File: MobilneSystemyInformatyczne/ShareWindsurfingEquipment/src/main/java/com/project/ShareWindsurfingEquipment/model/UserAccount.java
package com.project.ShareWindsurfingEquipment.model;

import com.project.ShareWindsurfingEquipment.common.Role;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table
public class UserAccount {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String login;

    @Column
    private String password;

    @Column
    private String email;

    @Column
    private String description;

    @Column
    private boolean instructor;

    @Column
    private BigDecimal pricePerHour;

    @Enumerated(EnumType.STRING)
    @Column
    private Role role;


    public UserAccount() {
    }

    public UserAccount(String login, String password, String email, String description, boolean instructor, BigDecimal pricePerHour, String role) {
        this.login = login;
        this.password = password;
        this.email = email;
        this.description = description;
        this.instructor = instructor;
        this.pricePerHour = pricePerHour;
        this.role = Role.valueOf(role);
    }


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public boolean isInstructor() {
        return instructor;
    }

    public void setInstructor(boolean instructor) {
        this.instructor = instructor;
    }

    public BigDecimal getPricePerHour() {
        return pricePerHour;
    }

    public void setPricePerHour(BigDecimal pricePerHour) {
        this.pricePerHour = pricePerHour;
    }

    public Role getRole() {
        return role;
    }

    public void setRole(Role role) {
        this.role = role;
    }

    @Override
    public String toString() {
        return "UserAccount{" +
                "id=" + id +
                ", login='" + login + '\'' +
                ", password='" + password + '\'' +
                ", email='" + email + '\'' +
                ", description='" + description + '\'' +
                ", instructor=" + instructor +
                ", pricePerHour=" + pricePerHour +
                ", role=" + role +
                '}';
    }
}
